### PR TITLE
QP-1722: Fix part 1: Update Function functions.

### PR DESCRIPTION
**Pull request by bzvestey**
https://maanainc.atlassian.net/browse/QP-1722

* Adds some comments to the Function functions
* Adds new functions for using service and name to load functions.

Screen Shot 2020-10-06 at 3 31 16 PM
Screen Shot 2020-10-06 at 3 32 37 PM
Screen Shot 2020-10-06 at 3 33 15 PM

### CODE REVIEW

File: src/AssistantAPIClient.js [hunk 1]
```diff
@@ -157,23 +159,127 @@ class AssistantAPIClient {
...
+  }
 
-  addFunctionExecutionListener = async (id, cb) => {
+  /**
+   * Adds a callback function to be called with the function has been executed.
+   *
+   * @param {string} id ID of the function.
+   * @param {function} cb The callback function.
+   */
+  addFunctionExecutionListener(id, cb) {
```
Comment (mechaffin) [inline src/AssistantAPIClient.js:271]: Should this one also be deprecated? and then have a version using the name?

File: src/AssistantAPIClient.js [hunk 2]
```diff
@@ -157,23 +159,127 @@ class AssistantAPIClient {
...
 
-  removeFunctionExecutionListener = async (id, cb) => {
+  /**
+   * Removes one or all callbacks listening for the function to be executed.
+   *
+   * @param {string} id ID of the function.
+   * @param {function} [cb] The callback function, if not supplied all of them
+   * are removed.
+   */
+  removeFunctionExecutionListener(id, cb) {
```
Comment (mechaffin) [inline src/AssistantAPIClient.js:282]: Should this one also be deprecated? and then have a version using the name?

**Review by mechaffin — APPROVED** [review]

File: src/AssistantAPIClient.js (hunk 1 above)
Comment (bzvestey) [reply src/AssistantAPIClient.js:271]: I have an item on my list to figure out what is going to happen with this.  I will probably be doing something like that.

File: src/AssistantAPIClient.js (hunk 2 above)
Comment (bzvestey) [reply src/AssistantAPIClient.js:282]: yeah, I have an item on list for handling function and graphql execution items.

**Pull request merged**